src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java

```diff
@@ -181,51 +99,49 @@ private void garbageCollector() {
 
         for (SapsImage task : failedTasks)
             deleteAllTaskFilesFromDisk(task);
-
     }
 
     /**
-     * This function gets tasks in failed state in Catalog.
+     * It gets tasks in failed state in {@code Catalog}.
      *
-     * @return tasks in specific state
+     * @return {@code SapsImage} list in {@code ImageTaskState.FAILED} state
      */
     private List<SapsImage> tasksInFailedState() {
         return CatalogUtils.getTasks(catalog, ImageTaskState.FAILED,
                 "gets tasks with " + ImageTaskState.FAILED.getValue() + " state");
     }
 
     /**
-     * This function cleans unfinished data from incomplete archiving.
+     * It cleans unfinished data from incomplete {@code ImageTaskState.ARCHIVING}.
      *
-     * @throws Exception
+     * @throws ArchiverException
      */
-    private void cleanUnfinishedArchivedData() throws Exception {
+    private void cleanUnfinishedArchivedData() throws ArchiverException {
         List<SapsImage> archivingTasks = tasksInArchivingState();
 
         LOGGER.info("Rollback in " + archivingTasks.size() + " tasks in archiving state");
 
         for (SapsImage task : archivingTasks) {
             LOGGER.info("Applying task [" + task.getTaskId() + "] rollback");
             rollBackArchive(task);
-            deletePendingFilesFromSwift(task);
+            permanentStorage.delete(task);
         }
     }
 
     /**
-     * This function gets tasks in finished state in Catalog.
+     * It gets tasks in archiving state in {@code Catalog}.
      *
-     * @return tasks in specific state
+     * @return {@code SapsImage} list in {@code ImageTaskState.ARCHIVING} state
```

Comment:
or empty in case of no archiving task, right?

Reply:
Done.